src/contracts/libraries/BeaconChainProofs.sol

```diff
@@ -103,29 +105,36 @@ library BeaconChainProofs {
 
 
     struct WithdrawalProofs {
+        bytes32 beaconStateRoot;
+        bytes latestBlockHeaderProof;
         bytes blockHeaderProof;
         bytes withdrawalProof;
         bytes slotProof;
         bytes executionPayloadProof;
-        bytes blockNumberProof;
+        bytes timestampProof;
         uint64 blockHeaderRootIndex;
         uint64 withdrawalIndex;
         bytes32 blockHeaderRoot;
         bytes32 blockBodyRoot;
         bytes32 slotRoot;
-        bytes32 blockNumberRoot;
+        bytes32 timestampRoot;
         bytes32 executionPayloadRoot;
     }
 
-    struct ValidatorFieldsAndBalanceProofs {
-        bytes validatorFieldsProof;
+    struct BalanceUpdateProofs {
```

Comment:
these should all be singular prob, not plural

Reply:
nah because the struct contains multiple different proofs?

Reply:
OK, so it's multiple different proofs, each proof proves a different thing about a singular validator.
pretty confusing, I do feel like a singular name might be less confusing, but don't feel strongly. all my other name suggestions at this point are word salad.

I would appreciate a top-level explanation on each of these struct definitions, at least though. Thinking just like one sentence, e.g. `@notice Collection of all of the proofs necessary to verify the effective balance of a single validator at a specific slot`

Reply:
done